nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-2-6-processors/src/main/resources/docs/org.apache.nifi.processors.kafka.pubsub.PublishKafkaRecord_2_6/additionalDetails.html

```diff
@@ -189,5 +189,77 @@ <h3>SASL_SSL</h3>
             See the SSL section for a description of how to configure the SSL Context Service based on the
             ssl.client.auth property.
         </p>
+        <h2>Record Key Output</h2>
+        <div>
+            <p>This processor includes optional properties that control the serialization of the key into the target
+                Kafka record.</p>
+            <ul>
+                <li>'Publish Strategy'</li>
+                <li>'Record Key Writer'</li>
+            </ul>
+
+            <p>'Publish Strategy' controls the mode used to convert the FlowFile record into a Kafka record.</p>
+            <ul>
+                <li>'Use Content as Record Value' (the default) persists the record key as a byte array.</li>
+                <li>'Use Wrapper' persists the record key as a record.</li>
+            </ul>
+
+            <p>If Publish Strategy 'Use Wrapper' is enabled, an additional processor configuration property is
+                activated: 'Record Key Writer'.</p>
+
+            <p>'Record Key Writer', if enabled, serializes the NiFi record key field into the Kafka record using the
+                specified implementation of the NiFi 'RecordSetWriter' interface. This may be used to emit the key field
+                as JSON, Avro, XML, or some other data format, where it may be used in the decision-making process of
+                downstream data processing (including that available in ConsumeKafkaRecord_2_6). If not defined, the
+                default implementation serializes the record key as an unmodified byte array (unchanged from previous
+                versions of the processor).</p>
+
+<p>Here is an example of a Kafka Record that is emitted by <code>JsonRecordSetWriter</code> when strategy "Use Wrapper" is active:</p>
```

Comment:
I think this section is inaccurate. This describes how the processor would behave if the "Publish Strategy" is set to "Use Content as Record Value", not "Use Wrapper".
I do think this is as good example, as it illustrates how a key can be specified, etc. But we should also have an example of how to use the Publish Strategy of "Use Wrapper". In this case, we wouldn't specify the Key Field, and the FlowFile content would be expected to have content that matches the wrapper's format. E.g.,
```
{
  "key": {"name":"Acme","number":"AC1234"},
  "value": {"address":"1234 First Street","zip":"12345","account":{"name":"Acme","number":"AC1234"}},
  "headers": {}
}
```

Reply:
It also is important here to note that the `metadata` field is currently ignored, if it is.
However, I don't think we should actually ignore the field. Or, at least, we should optionally ignore it. It would make sense to use both the `topic` and the `partition` values of the metadata fields. This would allow us to send in data that goes to multiple topics, even as a single transaction, which would be a pretty nice feature.
It's okay to ignore it for now, as long as the documentation very clearly & explicitly states that we ignore that field.

Reply:
Updated wrapper record to include metadata fields topic and timestamp.  If I pulled partition up to the beginning of the method, that could be included as well, but I don't know whether that would affect the application of the partition strategy.